oracle/store.go

```diff
@@ -0,0 +1,53 @@
+package oracle
+
+import (
+	"errors"
+	"sync"
+)
+
+// ErrCursorOutOfRange is returned when an iterator cursor is used to read a
+// value outside the range of the iterator.
+var ErrCursorOutOfRange = errors.New("cursor out of range")
+
+// MidpointPriceStorer is used for retrieving/storing MidpointPrice.
+type MidpointPriceStorer interface {
+
+	// PutMidpointPrice stores the given midPointPrice into the storer.
+	// It doesn't do any nonce check and will always overwrite previous record.
+	PutMidpointPrice(MidpointPrice) error
+
+	// MidpointPrice returns the latest mid-point price of the given token.
+	MidpointPrice() (MidpointPrice, error)
+}
+
+// midpointPriceStorer implements MidpointPriceStorer interface with an
+// in-memory map implementation. Data will be lost every time it restarts.
+type midpointPriceStorer struct {
```

Comment:
This concrete implementation is a driver, and belongs in the `leveldb` package more-so than the `oracle` core service package. It is still somewhat out of place in `leveldb` since it is an in-memory db, but in future refactors `leveldb` will be renamed to `db` regardless.